dev/MRTCore/mrt/Core/src/MRM.cpp

```diff
@@ -899,24 +903,38 @@ STDAPI MrmGetFilePathFromName(_In_ PCWSTR filename, _Outptr_ PWSTR* filePath)
         pointerToPath++;
     }
 
+    pointerToPath = pathAllocated ? pathAllocated.get() : path;
+
+    wchar_t moduleFilename[MAX_PATH];
+    RETURN_IF_FAILED(StringCchCopyW(moduleFilename, ARRAYSIZE(moduleFilename), lastSlash != nullptr ? lastSlash + 1 : pointerToPath));
+
     if (lastSlash != nullptr)
     {
         *(lastSlash + 1) = 0;
     }
 
-    pointerToPath = pathAllocated ? pathAllocated.get() : path;
-
     std::unique_ptr<wchar_t, decltype(&MrmFreeResource)> finalPath(nullptr, MrmFreeResource);
 
+    PCWSTR filenameToUse = filename;
+    if (filename == nullptr || *filename == L'\0')
+    {
+        filenameToUse = c_defaultPriFilename;
+    }
+
     // Will build the path at most twice.
-    // First time using current path. If not exist, do another time with parent path.
+    // If filename is provided:
+    //   - search under exe path
+    //   - if not exist, search parent path
+    // If filename is not provided:
+    //   - search under exe path with default name (resources.pri)
+    //   - if not exist, search under same path with [modulename].pri
```

Comment:
Why wouldn't we want to probe the parent directory if relying upon default filenames?

Reply:
i don't think we have a case where the pri file is created as [project name].pri and in the parent directory of exe.

Reply:
But `resources.pri` could be in the parent directory. I find it odd that a developer relying on the default behavior of MRT Core (by not specifying a filename) won't be able to handle one of the possible default behaviors of the rest of the stack (.wapproj putting the `resources.pri` in the parent directory). Am I misunderstanding how that scenario is intended to play out?

Reply:
for packaged app, "resources.pri" is passed into this API, and parent folder will be searched. The detail is encapsulated from the developer using ResourceManager.

Reply:
Got it, thanks for the explanation!